shotover-proxy/src/message/mod.rs

```diff
@@ -232,6 +232,32 @@ impl Message {
         self.invalidate_cache();
     }
 
+    pub fn set_backpressure(&mut self) {
+        *self = Message::from_frame(match &self.frame().unwrap() {
```

Comment:
Dont unwrap here either fail silently or log an error.

Also the current implementation of frame() is too expensive to call here.
There isnt really an alternative though.
We will have to either:
* rewrite the cassandra frame to not parse the CQL
* add a seperate method that just returns frame metadata that we can call instead of frame()
  + this should be even faster than the first suggestion so probably the best solution.

Reply:
Okay, I added the `fn metadata` method to `Message` in my latest commit. I've tried to make it extendable and so we can add other metadata types in the future, let me know if we would rather it just handle the Cassandra ones for now.